repairnator/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/notifier/TimedSummaryNotifier.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    /**
+     * Use the notifier engine to send an email to all addressess specified in the engine.
+     * @param subject the subject line of the message
+     * @param text the actual message
+     */
+    protected void notifyEngines(String subject, String text) {
+        for (NotifierEngine engine : this.engines) {
+            engine.notify(subject, text);
+        }
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+            if (this.intervalHasPassed()) {
+
+                MongoClient client = new MongoClient(
+                        new MongoClientURI(this.mongodbHost + "/" + this.mongodbName));
+                MongoDatabase mongo = client.getDatabase(this.mongodbName);
+
+                updateFilters(lastNotificationTime.getTime());
+
+                // Number of analyzed builds, rtscanner
+                Iterator<Document> nrOfDocuments = queryDatabase("rtscanner", rtscannerFilter, mongo);
+                int nrOfAnalyzedBuilds = nrOfObjects(nrOfDocuments);
+
+                // Number of repair attempts, inspector ComputeTestDir will most likely help
+
+                nrOfDocuments = queryDatabase("inspector", computeTestDirFilter, mongo);
+                int nrOfRepairAttempts = nrOfObjects(nrOfDocuments);
+
+                // Total number of patches, patches
+
+                nrOfDocuments = queryDatabase("patches", patchesFilter, mongo);
+                int nrOfPatches = nrOfObjects(nrOfDocuments);
+
+                // Number of patches per tool, patches with an if
+
+                int[] nrOfPatchesPerTool = new int[repairTools.length];
+
+                for (int i = 0; i < repairTools.length; i++) {
+                    nrOfDocuments = queryDatabase("patches", toolFilters.get(repairTools[i]), mongo);
+                    nrOfPatchesPerTool[i] = nrOfObjects(nrOfDocuments);
+                }
+                
+                Date now = new Date();
+                
+                String message = createMessage(nrOfAnalyzedBuilds, nrOfRepairAttempts, nrOfPatches, 
+                        nrOfPatchesPerTool, now);
+                updateCalendar(now);
+
+                notifyEngines("Repairnator: Summary email", message);
+
+                client.close();
+            } else {
+                try {
+                    Thread.sleep(TIME_TO_SLEEP);
+                } catch (InterruptedException e) {
+
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks whether the interval has passed since the last time a summary email was sent out.
+     * @return
+     */
+    protected boolean intervalHasPassed() {
+        return Duration.between(lastNotificationTime.getTime().toInstant(), (new Date()).toInstant())
+                .compareTo(this.interval) < 0;
+    }
+
+    /**
+     * Updates the lastNotificationTime
+     * @param newTime the latest time a notification was sent out
+     */
+    void updateCalendar(Date newTime) {
```

Comment:
Why didn't you name it updateLastNotificationTime?

Reply:
Did not think about it. Good idea, will change.